pkg/tests/tasks/observability/openshift_monitoring_test.go

```diff
@@ -0,0 +1,252 @@
+		smcpVer := env.GetSMCPVersion()
+		if smcpVer.LessThan(version.SMCP_2_4) {
+			t.Skip("integration with OpenShift Monitoring stack is not supported in OSSM older than v2.4.0")
+		}
+		smcpValues := map[string]string{
+			"Name":    smcpName,
+			"Version": smcpVer.String(),
+			"Member":  ns.Foo,
+		}
+
+		t.Cleanup(func() {
+			oc.DeleteFromString(t, meshNamespace, istiodServiceMonitor)
+			oc.DeleteFromString(t, ns.Foo, istioProxyMonitor)
+			oc.DeleteFromString(t, meshNamespace, enableTrafficMetrics)
+			oc.DeleteFromTemplate(t, monitoringNs, clusterMonitoringConfig, map[string]bool{"Enabled": false})
+			oc.DeleteFromString(t, meshNamespace, enableTrafficMetrics)
+			app.Uninstall(t, app.Httpbin(ns.Foo))
+			oc.DeleteFromTemplate(t, meshNamespace, mesh, smcpValues)
+		})
+
+		t.LogStep("Waiting until user workload monitoring stack is up and running")
+		oc.ApplyTemplate(t, monitoringNs, clusterMonitoringConfig, map[string]bool{"Enabled": true})
+		oc.WaitPodsExist(t, userWorkloadMonitoringNs)
+		oc.WaitAllPodsReady(t, userWorkloadMonitoringNs)
+
+		t.LogStep("Deploying SMCP")
+		oc.ApplyTemplate(t, meshNamespace, mesh, smcpValues)
+		oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+		t.LogStep("Enable Prometheus telemetry")
+		oc.ApplyString(t, meshNamespace, enableTrafficMetrics)
+
+		t.LogStep("Deploy httpbin")
+		app.InstallAndWaitReady(t, app.Httpbin(ns.Foo))
+		oc.WaitPodsExist(t, ns.Foo)
+		oc.WaitAllPodsReady(t, ns.Foo)
+
+		t.LogStep("Apply Prometheus monitors")
+		oc.ApplyString(t, meshNamespace, istiodServiceMonitor)
+		oc.ApplyString(t, ns.Foo, istioProxyMonitor)
+
+		t.LogStep("Generate some ingress traffic")
+		oc.ApplyFile(t, ns.Foo, "https://raw.githubusercontent.com/maistra/istio/maistra-2.4/samples/httpbin/httpbin-gateway.yaml")
+		httpbinURL := fmt.Sprintf("http://%s/headers", istio.GetIngressGatewayHost(t, meshNamespace))
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			curl.Request(t, httpbinURL, nil, assert.ResponseStatus(http.StatusOK))
+		})
+
+		t.LogStep("Fetch Thanos secret")
+		var secret string
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			secret = shell.Executef(t,
+				`kubectl get secrets --no-headers -o custom-columns=":metadata.name" -n %s | grep %s`, userWorkloadMonitoringNs, thanosTokenPrefix)
+			if strings.Contains(secret, "Error") {
+				t.Errorf("unexpected error: %s", secret)
+			}
+			// secret name contains '\n', because it's parsed from a single output column
+			secret = strings.TrimSuffix(secret, "\n")
+		})
+
+		t.LogStep("Fetch Thanos token")
+		var thanosToken string
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			thanosToken = shell.Executef(t, "oc get secret %s -n %s --template={{.data.token}} | base64 -d", secret, userWorkloadMonitoringNs)
+			if strings.Contains(thanosToken, "Error") {
+				t.Errorf("unexpected error: %s", thanosToken)
+			}
+		})
+
+		t.LogStep("Check istiod metrics")
+		checkMetricExists(t, meshNamespace, "pilot_info", thanosToken)
```

Comment:
Can we add here a t.Log with the expected, something like: "We should had from Thanos query metrics with this format: result":[{"metric":{"__name__":"pilot_info""